faker/providers/__init__.py

```diff
@@ -151,10 +151,15 @@ def random_number(self, digits=None, fix_len=False):
             fixed length number
         """
         if digits is None:
-            digits = self.random_digit()
+            digits = self.random_digit_not_null()
+        if digits < 0:
```

Comment:
Should also note that although it seems logical that this should be set to `if digits <= 0:`, this breaks test functionality for the python provider method for `pyfloat`.

Specifically this line: https://github.com/joke2k/faker/blob/16d2e6c343c476add06056d1a1929176f0c6acf2/tests/test_factory.py#L467